python/hail/dataset.py

```diff
@@ -2745,6 +2745,52 @@ def linreg_multi_pheno(self, ys, covariates=[], root='va.linreg', use_dosages=Fa
                                            jarray(Env.jvm().java.lang.String, covariates), root, use_dosages, min_ac,
                                            min_af)
         return VariantDataset(self.hc, jvds)
+    
+    @handle_py4j
+    @requireTGenotype
+    @typecheck_method(ys=listof(strlike),
+                      covariates=listof(strlike),
+                      root=strlike,
+                      use_dosages=bool,
+                      variant_block_size=integral)
+    def linreg3(self, ys, covariates=[], root='va.linreg', use_dosages=False, variant_block_size=16):
```

Comment:
Were you to keep the minAC / minAF filters (which I think are useful), I think you could replace linreg_multi_pheno without breaking interface. To do so, rather than re-using X across blocks, you could build the array of doubles that backs X using only those variants that pass filter, as well as an array of the indices of those variants within the block. Then at the end when you create annotations, you could overwrite null annotations with stats at those indices.